Code/Framework/AzCore/AzCore/Math/Vector2.inl

```diff
@@ -28,6 +32,16 @@ namespace AZ
         ;
     }
 
+    AZ_MATH_INLINE Vector2::Vector2(const Vector3& source)
+        : m_value(Simd::Vec3::ToVec2(source.GetSimdValue()))
```

Comment:
for SSE this is a copy of 128bits is there any problem with do this or would it be preferable to copy the individual components?

Reply:
This _seems_ legit to me, both the Vec2 and Vec3 are defined to be 128 bits for SSE. Is the specific concern that it's copying more bits than necessary? You could always throw together a quick benchmark to measure it both ways and see if there's any difference. My suspicion is that it's going to be identical or nearly identical, but I don't know for a fact.

Reply:
Could potentially run the benchmarks or make a new one https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/Tests/Math/Vector3PerformanceTests.cpp

Reply:
As long as scalar fallback still works correctly for those platforms that don't support a vector unit. On Arm that vector3 will be 128 bits and the vec2 will be only 64, but the convert/construct code here should be pretty optimal.